code/addons/docs/src/preset.ts

```diff
@@ -194,4 +194,12 @@ const docsX = docs as any;
 
 ensureReactPeerDeps();
 
-export { webpackX as webpack, indexersX as experimental_indexers, docsX as docs };
+const optimizeDeps = [
```

Comment:
@ndelangen I think we should move this to `framework.options.builder.optimizeDeps`, where other Vite-specific options are set too.

Reply:
why? that just overly complicates things, no?

Reply:
It just feels natural to locate this very Vite-specific option with the other Vite-specific options. From an API-design mindset I don't see why we would make it top-level, unless it was because we didn't intend this to be a public API.

Reply:
We decided to keep this API undocumented, and so we can make it a bit less "correct", and it is now just a top-level `optimizeViteDeps` config property.